wallet/src/main/scala/org/bitcoins/wallet/models/AddressTagTable.scala

```diff
@@ -0,0 +1,66 @@
+package org.bitcoins.wallet.models
+
+import org.bitcoins.core.protocol.BitcoinAddress
+import org.bitcoins.core.wallet.utxo.{
+  AddressTag,
+  AddressTagType,
+  InternalAddressTag,
+  UnknownAddressTag
+}
+import slick.jdbc.SQLiteProfile.api._
+import slick.lifted.ProvenShape
+
+case class AddressTagDb(
+    address: BitcoinAddress,
+    tag: AddressTag,
+    tagType: AddressTagType) {
+  require(
+    tag.tagType.name.isEmpty || tag.tagType == tagType,
+    s"If tag's tagType is defined it must be the same as tagType, got $tagType")
+
+  val unknownAddressTag: UnknownAddressTag =
+    UnknownAddressTag(tag.name, tagType)
```

Comment:
I'd prefer `internalAddressTag` which only returns `Unknown` if it is not an internal tag

Reply:
that is essentially what `val addressTag` is. With context of replies to above comments, I think it makes sense to do it this way